packages/shared/src/utils/browser.ts

```diff
@@ -61,3 +61,18 @@ export function isValidBrowserOnline(): boolean {
 
   return !isUserAgentRobot && !isWebDriver && isExperimentalConnectionOnline && isNavigatorOnline;
 }
+
+export function isBrowserOnline(): boolean {
```

Comment:
We already have an `isValidBrowserOnline` function. Can we refactor/reuse that?

Reply:
We want the opposite logic here. If it's not a valid browser, but it's online, we don't want to retry forever.

Reply:
We could split the two checks into two smaller utilities and use those.

Reply:
☝️ Let's do that and wrap this up.